Dockerfile

```diff
@@ -0,0 +1,21 @@
+# syntax=docker/dockerfile:1
+FROM python:3.7-slim-bullseye
+WORKDIR /
+RUN mkdir -p /opt/virtualenv
+ENV VIRTUAL_ENV="/opt/virtualenv/_installTop_"
+RUN python -m venv $VIRTUAL_ENV
+ENV PATH="$VIRTUAL_ENV/bin${PATH:+:}$PATH"
+COPY . $VIRTUAL_ENV/djaodjin-deployutils/
+WORKDIR $VIRTUAL_ENV/djaodjin-deployutils
+RUN set -eux;\
```

Comment:
Any reason why it is a single `RUN` line instead of one per command?

Reply:
1. For context when maintaining the Dockerfile.
2. To be sure that any compilation step or make recipe has it's `ENV` preserved.
3. To ensure that build-time dependencies installed via `apt` get removed in the same step. This isn't really an issue in this dockerfile, but can be in more complex dockerfiles. Apt does not have great tooling to manage this use case, and adding and removing build-deps can get kind of awkward; see [https://github.com/docker-library/python/blob/0a9ee3e64588bb1144d6e4e413a0c5dd5cd48651/3.7/slim-bullseye/Dockerfile]() for an example.

We've talked about running "make recipes" outside of docker containers already, but I think any make recipe that requires either managing dependencies, or requires libraries / tools that are tied to the runtime environment  contained within the docker image, should be run and cleaned up in the docker build phase. This helps with dependency tracking, bug resolution, and maintainability.